bundles/framework/coordinatetool/plugin/CoordinatePluginHandler.js

```diff
@@ -0,0 +1,625 @@
+                const toProjection = this.state.selectedProjection;
+                xy = this.coordinateTransformationExtension.transformCoordinates(xy, fromProjection, toProjection);
+            } catch (e) {}
+        }
+
+        const isSupported = !!((this.config && Array.isArray(this.config.supportedProjections)));
+        const isDifferentProjection = !!((this.state.selectedProjection !== this.mapModule.getProjection() &&
+            xy.lonlat.lat !== 0 && xy.lonlat.lon !== 0));
+        let lat = parseFloat(xy.lonlat.lat);
+        let lon = parseFloat(xy.lonlat.lon);
+
+        lat = lat + '';
+        lon = lon + '';
+        if (lat.indexOf('~') === 0) {
+            lat = lat.substring(1, lat.length);
+        }
+        lat = lat.replace(/,/g, '.');
+        if (lon.indexOf('~') === 0) {
+            lon = lon.substring(1, lat.length);
+        }
+        lon = lon.replace(/,/g, '.');
+
+        lat = lat + '';
+        lon = lon + '';
+        if (lat.indexOf('~') === 0) {
+            lat = lat.substring(1, lat.length);
+        }
+        lat = lat.replace(/,/g, '.');
+        if (lon.indexOf('~') === 0) {
+            lon = lon.substring(1, lat.length);
+        }
+        lon = lon.replace(/,/g, '.');
+
+        // Need to show degrees ?
+        if (this.allowDegrees() && !isNaN(lat) && !isNaN(lon)) {
+            const degreePoint = Oskari.util.coordinateMetricToDegrees([lon, lat], this.getProjectionDecimals());
+            lon = degreePoint[0];
+            lat = degreePoint[1];
+        }
+        // Otherwise show meter units
+        else if (!isNaN(lat) && !isNaN(lon)) {
+            lat = parseFloat(lat);
+            lon = parseFloat(lon);
+            lat = lat.toFixed(this.getProjectionDecimals());
+            lon = lon.toFixed(this.getProjectionDecimals());
+            lat = this.formatNumber(lat, this.decimalSeparator);
+            lon = this.formatNumber(lon, this.decimalSeparator);
+        }
+
+        // Not from server
+        if (isSupported && isDifferentProjection && !fromServer) {
+            lat = '~' + lat;
+            lon = '~' + lon;
+        }
+
+        if (fromServer) {
+            if (lon.indexOf('~') >= 0) {
+                lon = lon.replace('~', '');
+            }
+            if (lat.indexOf('~') >= 0) {
+                lat = lat.replace('~', '');
+            }
+        }
+
+        this.updateState({
+            displayXy: {
+                'lonlat': {
+                    'lon': lon,
+                    'lat': lat
+                }
+            },
+            loading: false
+        });
+
+        this.updatePopup();
+    }
+
+    setSelectedProjection (projection) {
+        this.updateState({
+            selectedProjection: projection
+        });
+        this.updateLonLat();
+    }
+
+    showErrorMessage (message, title) {
+        const dialog = Oskari.clazz.create('Oskari.userinterface.component.Popup');
```

Comment:
Let's move these error messages from jQuery popup to using Messaging.error() or showing the error message next/close to the field (whichever makes more sense for the error).